src/Common/tests/System/Collections/IList.NonGeneric.Tests.cs

```diff
@@ -863,19 +865,34 @@ public void IList_NonGeneric_Remove_NullNotContainedInCollection(int count)
         [MemberData(nameof(ValidCollectionSizes))]
         public void IList_NonGeneric_Remove_NonNullNotContainedInCollection(int count)
         {
-            if (!IsReadOnly && !ExpectedFixedSize)
+            if (!IsReadOnly && !ExpectedFixedSize && !IList_NonGeneric_RemoveNonExistent_Throws)
             {
                 int seed = count * 251;
-                IList collection = NonGenericIListFactory(count);
+                IList list = NonGenericIListFactory(count);
                 object value = CreateT(seed++);
-                while (collection.Contains(value) || Enumerable.Contains(InvalidValues, value))
+                while (list.Contains(value) || Enumerable.Contains(InvalidValues, value))
                     value = CreateT(seed++);
-                collection.Remove(value);
-                Assert.Equal(count, collection.Count);
+                list.Remove(value);
+                Assert.Equal(count, list.Count);
             }
-        }
-
-        [Theory]
+		}
+
+		[Theory]
+		[MemberData(nameof(ValidCollectionSizes))]
+		public void IList_NonGeneric_Remove_NonNullNotContainedInCollection_Throws(int count)
+		{
+			if (IList_NonGeneric_RemoveNonExistent_Throws)
```

Comment:
Same comment as earlier.  I'd prefer we not introduce a new test for this and just add a conditional to the previous one, e.g.
```C#
if (IList_NonGeneric_RemoveNonExistent_Throws)
{
    Assert.Throws<ArgumentException>(() => list.Remove(value));
}
else
{
    list.Remove(value);
    Assert.Equal(count, list.Count);
}
```
We otherwise end up significantly increasing the number of test cases, but half of them end up being nops for a given collection.